consensus/consensus-types/src/common.rs

```diff
@@ -285,9 +285,10 @@ impl Payload {
         match (quorum_store_enabled, self) {
             (false, Payload::DirectMempool(_)) => Ok(()),
             (true, Payload::InQuorumStore(proof_with_status)) => {
-                for proof in proof_with_status.proofs.iter() {
-                    proof.verify(validator)?;
-                }
+                proof_with_status
+                    .proofs
+                    .par_iter()
+                    .try_for_each(|proof| proof.verify(validator))?;
```

Comment:
we may want to chunk them instead of every single one, spawn task is non trivial as well

Reply:
+1, rayon spawn is very expensive - especially done on small item. Use a min size with the parallel item.